plugin/completion/lib_complete.py

```diff
@@ -198,6 +203,115 @@ def complete(self, completion_request):
         log.debug(' completions: %s' % completions)
         return (completion_request, completions)
 
+    @staticmethod
+    def location_from_type(clangType):
+        """ Return location from type.
+
+            Return proper location from type.
+            Remove all inderactions like pointers etc.
+
+        """
+        cursor = clangType.get_declaration()
+        if cursor and cursor.location and cursor.location.file:
+            return cursor.location
+
+        cursor = clangType.get_pointee().get_declaration()
+        if cursor and cursor.location and cursor.location.file:
+            return cursor.location
+
+        return None
+
+    @staticmethod
+    def link_from_location(location, text):
+        """Provide link to given cursor.
+
+            Transforms SourceLocation object into html string
+
+        """
+        result = ""
+        if location and location.file and location.file.name:
+            result += "<a href=\""
+            result += location.file.name
+            result += ":"
+            result += str(location.line)
+            result += ":"
+            result += str(location.column)
+            result += "\">" + text + "</a>"
+        else:
+            result += text
+        return result
+
+    def build_info_details(self, cursor):
+        """Provide information about given cursor.
+
+        Builds detailed information about cursor.
+
+        """
+        result = ""
+        if cursor.result_type.spelling:
```

Comment:
Can we just do smth like `cursor_type = cursor.result_type` if `spelling` is present for it and `cursor_type = cursor.type` otherwise? This will reduce the `if` and reduce duplication.